code/src/components/Form.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from "react";
+import Question from "components/Question";
+import Button from "components/Button";
+import SubmittedView from "components/SubmittedView";
+import "./Form.css";
+
+const questions = [
+  {
+    id: "preference",
+    question_text: "What is your dietary preference",
+    input_type: "select",
+    options: [
+      { value: "Eat everything", label: "Eat everything" },
+      { value: "Pescatarian", label: "Pescatarian" },
+      { value: "Vegetarian", label: "Vegetarian" },
+      { value: "Vegan", label: "Vegan" },
+    ],
+  },
+  {
+    id: "cuisine",
+    question_text: "Which of the following cuisines are you craving most?",
+    input_type: "radio",
+    options: [
+      { value: "Persian cuisine", label: "Persian" },
+      { value: "Mexican cuisine", label: "Mexican" },
+      { value: "Japanese cuisine", label: "Japanese" },
+      { value: "Any cuisine", label: "Anything goes" },
+    ],
+  },
+  {
+    id: "spiciness",
+    question_text: "How well can you handle spice?",
+    input_type: "range",
+    format_answer: (value) => `Spice level: ${value} ${Number(value) === 1 ? "chili" : "chilis"}`,
+    options: [
+      { value: 1, label: "\u{1F336}" },
+      { value: 2, label: "\u{1F336}\u{1F336}" },
+      { value: 3, label: "\u{1F336}\u{1F336}\u{1F336}" },
+    ],
+  },
+  {
+    id: "herbs",
+    question_text: "Which of these herbs is superior?",
+    input_type: "select",
+    format_answer: (value) => `Heaps of ${value}`,
+    options: [
+      { value: "coriander", label: "Coriander" },
+      { value: "basil", label: "Basil" },
+      { value: "mint", label: "Mint" },
+      { value: "dill", label: "Dill" },
+      { value: "parsley", label: "Parsley" },
+    ],
+  },
+  {
+    id: "allergies",
+    question_text: "Do you have an allergies we should know about?",
+    input_type: "text",
+    format_answer: (value) => `Allergies: ${value.charAt(0).toUpperCase() + value.slice(1)}`,
+  },
+];
+
+const Form = () => {
+  const [currentQuestionIndex, setQuestionIndex] = useState(0);
+  const [answers, setAnswers] = useState([]);
+  const [inputValue, setInputValue] = useState("");
+
+  const buttonLabel = currentQuestionIndex === questions.length - 1 ? "Submit" : "Next question";
+
+  const handleInputChange = (event) => setInputValue(event.target.value);
+
+  const formatInput = (value) =>
+    questions[currentQuestionIndex].format_answer
+      ? questions[currentQuestionIndex].format_answer(value)
+      : value;
+
+  const handleOnSubmit = (event) => {
+    event.preventDefault();
+    setAnswers((arr) => [...arr, formatInput(inputValue)]);
+    setQuestionIndex(currentQuestionIndex + 1);
+    setInputValue("");
+  };
+
+  const didSubmit = () => questions.length === answers.length;
+
+  const resetForm = () => {
+    setAnswers([]);
+    setQuestionIndex(0);
+  };
+
+  if (didSubmit()) {
+    return <SubmittedView answers={answers} reset={resetForm} />;
```

Comment:
Since <SubmittedView /> is stateless, I'd be pretty ok with its content living here.

Reply:
Question here: I have a hard time visualising this, when you say the content living here, do you mean that `<SubmittedView />` as a component is not needed, ex it could be rendered like so:
```javascript
if (didSubmit()) {
  return (
    <section className="container">
      <h2 className="submit-message">Awesome! Our next dinner party will be designed around your preferences: </h2>
      <ul className="submit-list">
        {answers.map((answer, i) => (
          <li className="submit-list-item" key={i}>{answer}</li>
        ))}
      </ul>
      <button onClick={reset}>Reset</button>
    </section>
  );  
} 
```
or do you mean defining `<SubmittedView />` in the same file as `<Form />` in Form.js. I don't see how that would be different apart from the need to import and export, I'm just unsure... ex:
```javascript
//Form.js 

const Form = () => {
  return (
    //...
    <SubmittedView />
  )
};

const SubmittedView = () => {
  return //...
};
```
More importantly, whichever you mean - why would you be ok with the content in that way? I think I'm missing an understanding of how to work with stateless vs stateful components since I haven't considered it really, so I can look into that.

Reply:
Exactly! My initial thought was that it would look like example 1 or 2. The reasoning behind that is that if a component is only consumed in **one place**, maybe it doesn't need to be its own independent component. 

`<SubmittedView />` Could be considered a [dumb component](https://medium.com/@thejasonfile/dumb-components-and-smart-components-e7b33a698d43#:~:text=Dumb%20components%20are%20also%20called,see%20how%20things%20are%20going.)  in a good way. In the case of going down the route of dividing your components into smart and dumb components maybe a folder structure could help visualising that: `./src/components/<smart components goes here>` and `./src/view/<dumb components goes here>`. 

I forgot to also say, take my recommendations with a gram of salt. It's completely fine to disagree with me. At the end of the day it's important that you have an architecture that makes sense to **you**. Thanks for asking questions thought, that's what we're here for! :)

Reply:
Ahh, ok great explanation! I see what you mean with smart vs dumb.

Since it's only the third week in React I consider myself more of a sponge trying to absorb information around me, rather than having strong opinions myself. That time will come... :) Thanks again!